### DISCUSSION TITLE
Feature Request: Ability to visually hide certain fields in blocks.

### DISCUSSION THREAD

**Discussion by richardvanbergen** [payloadcms/payload#125] [Feature Requests & Ideas]
# Feature Request

I would like the ability to either:

- Collapse/Expand groups similar to Sanity's [collapsible](https://www.sanity.io/docs/object-type#collapsible-555aeff73448) attribute on fieldsets
- Organise groups into tabs on a given block
- Provide a separate "block settings" UI for certain properties

What the best approach to take I'm not sure but ideally it shouldn't change the shape of the data coming back from the API, it'd just be a way of visually organising repeated properties, similar to the `admin: { sidebar: true }` property.

## Use-Case

A really nice feature is being able to visually stash of fields in the sidebar, which means that editors can focus on content in the main area. Now, some of my blocks have some shared optional groups, things like background properties. If I type and test things properly this means I can make the block editing experience super flexible without increasing the complexity of the code via ✨**composition**✨.

The main problem though is that it clutters up the editing experience a lot as seen in the screenshot attached. 🙏

## Additional Detail

![image](https://user-images.githubusercontent.com/3854592/107166147-72936400-69f8-11eb-8cdc-ab3ef63e8cd3.png)

**Comment by jmikrut** [comment]
Hey @richardvanbergen — thank you for your continued contributions! This is great. Keep it coming.

A few of these feature requests are actually already on our roadmap - specifically, a `Tabs` field and a `Collapsible` field. They'll operate similar to the `Row`—not changing data and only impacting the admin panel just as you've described. 

Our roadmap will be public shortly - but until then, stay tuned for more. 

Also, looks liike you're making great use of many of Payload's features. Awesome to see.

> Provide a separate "block settings" UI for certain properties

Can I ask for a bit more info here? What types of block settings would you like to see?

**Comment by richardvanbergen** [comment]
Hi @jmikrut This is the first CMS that I've actually **enjoyed** working with and believe me when I say -- I've tried a lot of them. So don't worry, you'll be sick of seeing my name pop up in your inbox by the time I'm done. 😉

> Can I ask for a bit more info here? What types of block settings would you like to see?

I wasn't talking about any intrinsic block settings because I think it's easy enough to add your own anyway. I was just waxing lyrical about potential UI elements, maybe a little cog underneath the expand ([+]) button which would then toggle a settings UI which is distinct from the normal content UI.
